src/delay.rs

```diff
@@ -1,87 +1,30 @@
 //! Delay devices and providers
 use crate::register::mcycle;
-use embedded_hal::blocking::delay::{DelayMs, DelayUs};
+use embedded_hal::delay::DelayUs;
 
 /// Machine mode cycle counter (`mcycle`) as a delay provider
 #[derive(Copy, Clone)]
+#[repr(transparent)]
```

Comment:
What's the purpose of `#[repr(transparent)]` here?

Reply:
Not sure, to be honest. As far as I know, it is desirable for FFI with C to avoid potential weird compiler allocations. But maybe it is useless, I can remove it.

Reply:
I don't think it is going to cause any harm lets leave it as is.